src/callback_scope.rs

```diff
@@ -1,70 +1,76 @@
-use crate::scope::{Scope, Scoped};
-use crate::Context;
-use crate::InIsolate;
+use std::mem::MaybeUninit;
+
+use crate::scope::Entered;
+use crate::scope::Scope;
+use crate::scope::Scoped;
+use crate::scope_traits::internal::GetRawIsolate;
+use crate::FunctionCallbackInfo;
 use crate::Isolate;
 use crate::Local;
-use crate::Message;
-use crate::Promise;
 use crate::PromiseRejectMessage;
-use std::mem::MaybeUninit;
-
-extern "C" {
-  fn v8__Promise__GetIsolate(promise: *mut Promise) -> *mut Isolate;
-}
+use crate::PropertyCallbackInfo;
 
-pub trait GetIsolate
-where
-  Self: Sized,
-{
-  fn get_isolate(&mut self) -> &mut Isolate;
+/// A CallbackScope can be used to obtain a mutable Isolate reference within
+/// a callback that is called by V8 on the thread that already has a Locker
+/// on the stack.
+///
+/// Using a CallbackScope in any other situation is unsafe.
+/// Also note that CallbackScope should not be used for function and property
+/// accessor callbacks; use FunctionCallbackScope and PropertyCallbackScope
+/// instead.
+///
+/// A CallbackScope can be created from the following inputs:
```

Comment:
This comment is actually not so relevant any more.
I entend to pass a suitable Scope object to all callbacks that need one. The user won't have to create one manually.